app/worker.php

```diff
@@ -72,61 +113,150 @@
-}, ['register']);
+/**
+ * Get Files Storage Device
+ * @param string $projectId of the project
+ * @return Device
+ */
+Server::setResource('getFilesDevice', function () {
+    return function (string $projectId) {
+        return getDevice(APP_STORAGE_UPLOADS . '/app-' . $projectId);
+    };
+});
 
-Server::setResource('pools', function ($register) {
-    return $register->get('pools');
-}, ['register']);
+/**
+ * Get Builds Storage Device
+ * @param string $projectId of the project
+ * @return Device
+ */
+Server::setResource('getBuildsDevice', function () {
+    return function (string $projectId) {
+        return getDevice(APP_STORAGE_BUILDS . '/app-' . $projectId);
+    };
+});
+
+/**
+ * Get cache  Device
+ * @param string $projectId of the project
+ * @return Device
+ */
+Server::setResource('getCacheDevice', function () {
+    return function (string $projectId) {
+        return getDevice(APP_STORAGE_CACHE . '/app-' . $projectId);
+    };
+});
 
 $pools = $register->get('pools');
-$connection = $pools->get('queue')->pop()->getResource();
-$workerNumber = swoole_cpu_num() * intval(App::getEnv('_APP_WORKER_PER_CORE', 6));
+$platform = new Appwrite();
+$args = $_SERVER['argv'];
 
-if (empty(App::getEnv('QUEUE'))) {
-    throw new Exception('Please configure "QUEUE" environment variable.');
+if (!isset($args[1])) {
+    Console::error('Missing worker name');
+    Console::exit(1);
 }
 
-$adapter = new Swoole($connection, $workerNumber, App::getEnv('QUEUE'));
-$server = new Server($adapter);
+\array_shift($args);
+$workerName = $args[0];
+$workerIndex = $args[1] ?? '';
 
-$server
+if (!empty($workerNum)) {
+    $workerName .= '_' . $workerIndex;
+}
+
+try {
+    /**
+     * Any worker can be configured with the following env vars:
+     * - _APP_WORKERS_NUM           The total number of worker processes
+     * - _APP_WORKER_PER_CORE       The number of worker processes per core (ignored if _APP_WORKERS_NUM is set)
+     * - _APP_QUEUE_NAME  The name of the queue to read for database events
+     */
+    $platform->init(Service::TYPE_WORKER, [
+        'workersNum' => App::getEnv('_APP_WORKERS_NUM', swoole_cpu_num() * intval(App::getEnv('_APP_WORKER_PER_CORE', 6))),
+        'connection' => $pools->get('queue')->pop()->getResource(),
+        'workerName' => strtolower($workerName) ?? null,
+        'queueName' => App::getEnv('_APP_QUEUE_NAME', 'v1-' . strtolower($workerName))
+    ]);
+} catch (\Exception $e) {
+    Console::error($e->getMessage() . ', File: ' . $e->getFile() .  ', Line: ' . $e->getLine());
+}
+
+$worker = $platform->getWorker();
+
+$worker
     ->shutdown()
     ->inject('pools')
     ->action(function (Group $pools) {
         $pools->reclaim();
     });
 
-$server
+$worker
     ->error()
     ->inject('error')
     ->inject('logger')
     ->inject('log')
-    ->action(function (Throwable $error, ?Logger $logger, Log $log) {
+    ->action(function (Throwable $error, Logger|null $logger, Log $log) {
```

Comment:
```suggestion
    ->action(function (Throwable $error, ?Logger $logger, Log $log) {
```
What is the difference between the two?